### PR TITLE
Backport b4d39830

### PR DESCRIPTION
**Pull request by SemenchenkoVitaliy**
Original commit message:
  Fix serving application folder index

  Co-authored-by: tshemsedinov <timur.shemsedinov@gmail.com>
  PR-URL: https://github.com/metarhia/impress/pull/1124

Refs: https://github.com/metarhia/impress/commit/b4d39830f27476c61cd452d3187abc13f3eb1846
Closes: https://github.com/metarhia/impress/issues/1126

/cc @tshemsedinov @nechaido @belochub @lundibundi.

### CODE REVIEW

File: lib/impress.client.js [hunk 1]
```diff
@@ -936,10 +936,13 @@ Client.prototype.static = function(
...
 
-  const isRoot = client.path === '/';
-  let relPath = '/static';
-  if (isRoot) relPath += '/index.html';
-  else relPath += api.querystring.unescape(client.url);
+  const basePath = '/static';
+  let relPath = '/static/index.html';
+  if (client.path !== '/') {
+    const url = api.querystring.unescape(client.url);
+    const safePath = api.path.join(basePath, url);
```
Comment (nechaido) [inline lib/impress.client.js]:
```suggestion
    const safePath = api.path.resolve(basePath, url);
```

**Comment by tshemsedinov** [comment]
CI fails

**Comment by SemenchenkoVitaliy** [comment]
@tshemsedinov It seems that this is a problem with the `mongodb.provider` at `globalstorage`. More on that, previous CI for 0.5.2 failed with the same error: https://travis-ci.org/metarhia/impress/builds/371805434

**Comment by belochub** [comment]
@SemenchenkoVitaliy to fix the CI build, you can revert https://github.com/metarhia/impress/commit/a051d8a29310a3653b2a0baadfec93fdb89fb778 here.

**Pull request closed without merging**